plugins/chain_api_plugin/chain_api_plugin.cpp

```diff
@@ -703,75 +704,74 @@ static float64_t to_softfloat64( double d ) {
+        const bool is_active = value["commencement_block"].as_uint64() != 0;
+
+        auto producer_info = fc::mutable_variant_object()("owner", account)
+                                                         ("total_votes", votes)
+                                                         ("producer_key", key)
+                                                         ("url", "");
+
+        if (is_active) {
+            producer_info["is_active"] = 1;
+        }
+        rows.emplace_back(producer_info);
+        ++count;
+    }
+
+    return {rows, total_votes, next_producer};
+
+
+    if (p.lower_bound.empty()) {
+    } else {
+        chaindb.lower_bound(request, fc::mutable_variant_object()("name", p.lower_bound));
+    }
 }
 
-get_producers_result chain_api_plugin_impl::get_producers( const get_producers_params& p ) const {
-//   const chain::abi_def abi = eosio::get_abi(chain_controller_, chain::config::system_account_name);
-//   const auto table_type = get_index_type(abi, N(producers));
-//   const chain::abi_serializer abis{ abi, abi_serializer_max_time_ };
- //  EOS_ASSERT(table_type == KEYi64, chain::contract_table_query_exception, "Invalid table type ${type} for table producers", ("type",table_type));
+std::string chain_api_plugin_impl::get_agent_public_key(chain::account_name account) const {
+    auto& chaindb = chain_controller_.chaindb();
 
-// TODO: Removed by CyberWay
-//   const auto& d = db_controller_.db();
-//   const auto lower = name{p.lower_bound};
-//
-//   static const uint8_t secondary_index_num = 0;
-//   const auto* const table_id = d.find<chain::table_id_object, chain::by_code_scope_table>(
-//           boost::make_tuple(config::system_account_name, config::system_account_name, N(producers)));
-//   const auto* const secondary_table_id = d.find<chain::table_id_object, chain::by_code_scope_table>(
-//           boost::make_tuple(config::system_account_name, config::system_account_name, N(producers) | secondary_index_num));
-//   EOS_ASSERT(table_id && secondary_table_id, chain::contract_table_query_exception, "Missing producers table");
-//
-//   const auto& kv_index = d.get_index<key_value_index, by_scope_primary>();
-//   const auto& secondary_index = d.get_index<index_double_index>().indices();
-//   const auto& secondary_index_by_primary = secondary_index.get<by_primary>();
-//   const auto& secondary_index_by_secondary = secondary_index.get<by_secondary>();
+    const cyberway::chaindb::index_request request{N(), N(), N(stake.agent), N(bykey)};
 
-   get_producers_result result;
-   const auto stopTime = fc::time_point::now() + fc::microseconds(1000 * 10); // 10ms
-   std::vector<char> data;
+    const auto it = chaindb.lower_bound(request, fc::mutable_variant_object()("token_code", chain::symbol(CORE_SYMBOL).to_symbol_code())("account", account));
+
+    if (it.pk == cyberway::chaindb::end_primary_key) {
+        EOS_THROW(cyberway::chaindb::chaindb_exception,
```

Comment:
This throw doesn't allow to get information about all other producers.

You shouldn't break loading information on errors in data.
It make API not-useful.